File: jft.academia.v01/Engine/jft.academia.v01.Aplicativo/DataViews/RepositorysViews/AtividadesRepositoryView.cs
using jft.academia.v01.Aplicativo.Common;
using jft.academia.v01.Aplicativo.ModelsEntities;
using jft.academia.v01.Core.Entities;
using jft.academia.v01.Core.EntitiesViews;
using jft.academia.v01.Core.Entities.Interfaces;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

//namespace jft.academia.v01.Aplicativo.DataViews.RepositorysViews
//{
//    public class AtividadesRepositoryView : 
//        BasePageModelView<Atividades, AtividadesModel>, 
//        IDataStoreViewAsync<Atividades, AtividadesModel>
//    {

        



//        public Task<bool> DeleteViewItemAsync(AtividadesModel item)
//        {
//            throw new NotImplementedException();
//        }

//        public Task<AtividadesModel> GetEntityToModelAsync(Atividades _entity)
//        {
//            var _r = new AtividadesModel
//            {
//                id_atividade = _entity.id_atividade,
//                nm_atividade = _entity.nm_atividade,
//                te_descricao = _entity.te_descricao,

//            };

//            return Task.FromResult(_r);

//        }

//        public Task<Atividades> GetModelToEntityAsync(AtividadesModel _entityModel)
//        {
//            throw new NotImplementedException();
//        }

//        public Task<AtividadesModel> GetViewItemAsync(int id)
//        {
//            var _result = this.dbAcademia.Atividades.GetItemAsync(id).Result;


//            return this.GetEntityToModelAsync(_result);
//        }

//        public Task<List<AtividadesModel>> GetViewItemsAsync(bool forceRefresh = false)
//        {
             

//            var _result = this.dbAcademia.Atividades.GetItemsAsync().ContinueWith(t =>
//            {
                
//                var atividades = t.Result;
//                var listAtividadesView = new List<AtividadesModel>();
                
//                foreach (var _item in atividades)
//                {
//                    var _view = this.GetEntityToModelAsync(_item).Result;
//                    listAtividadesView.Add(_view);
//                }
//                return listAtividadesView;
//            });


//            return _result;


//        }

//        public Task<bool> SaveViewItemAsync(AtividadesModel item)
//        {
//            throw new NotImplementedException();
//        }






//    }
//}

File: jft.academia.v01/Engine/jft.Academia.v01.Infra01/DataViews/RepositorysViews/AtividadesRepositoryView.vb
Imports jft.Academia.v01.Core.Entities
Imports jft.Academia.v01.Core.Entities.Interfaces
Imports jft.Academia.v01.Core.EntitiesViews
Imports jft.Academia.v01.Infra01.DataViews.Common

Namespace DataViews



    Namespace RepositorysViews



        Public Class AtividadesRepositoryView
            Inherits BaseRepositoryView(Of Atividades, AtividadesModel)
            Implements IDataStoreViewAsync(Of Atividades, AtividadesModel)



            'Public Function DeleteViewItemAsync(item As AtividadesModel) As Task(Of Boolean) Implements IDataStoreViewAsync(Of Atividades, AtividadesModel).DeleteViewItemAsync
            '    Throw New NotImplementedException()
            'End Function

            Public Overloads Function GetEntityToModelAsync(_entity As Atividades) As Task(Of AtividadesModel) Implements IDataStoreViewAsync(Of Atividades, AtividadesModel).GetEntityToModelAsync

                Dim _r = New AtividadesModel With {
                    .id_atividade = _entity.id_atividade,
                    .nm_atividade = _entity.nm_atividade,
                    .te_descricao = _entity.te_descricao
                }

                Return Task.FromResult(_r)

            End Function

            'Public Function GetModelToEntityAsync(_entityModel As AtividadesModel) As Task(Of Atividades) Implements IDataStoreViewAsync(Of Atividades, AtividadesModel).GetModelToEntityAsync
            '    Throw New NotImplementedException()
            'End Function

            Public Overloads Function GetViewItemAsync(id As Integer) As Task(Of AtividadesModel) Implements IDataStoreViewAsync(Of Atividades, AtividadesModel).GetViewItemAsync

                Dim _result = Me.dbAcademia.Atividades.GetItemAsync(id).Result
                Return Me.GetEntityToModelAsync(_result)

            End Function

            Public Overloads Function GetViewItemsAsync(Optional forceRefresh As Boolean = False) As Task(Of List(Of AtividadesModel)) Implements IDataStoreViewAsync(Of Atividades, AtividadesModel).GetViewItemsAsync


                Dim _result = Me.dbAcademia.Atividades _
                    .GetItemsAsync().ContinueWith(Function(t)

                                                      Dim atividades = t.Result
                                                      Dim listAtividadesView = New List(Of AtividadesModel)()

                                                      For Each _item In atividades
                                                          Dim _view = GetEntityToModelAsync(CType(_item, Atividades)).Result
                                                          listAtividadesView.Add(_view)
                                                      Next
                                                      Return listAtividadesView
                                                  End Function)


                Return _result


            End Function

            'Public Function SaveViewItemAsync(item As AtividadesModel) As Task(Of Boolean) Implements IDataStoreViewAsync(Of Atividades, AtividadesModel).SaveViewItemAsync
            '    Throw New NotImplementedException()
            'End Function

        End Class

    End Namespace

End Namespace